pkg/app/pagewriter/static_pages.go

```diff
@@ -0,0 +1,92 @@
+package pagewriter
+
+import (
+	// Import embed to allow importing default page templates
+	_ "embed"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/logger"
+)
+
+const (
+	robotsTxtName = "robots.txt"
+)
+
+//go:embed robots.txt
+var defaultRobotsTxt []byte
+
+// staticPageWriter is used to write static pages.
+type staticPageWriter struct {
+	pages           map[string][]byte
+	errorPageWriter *errorPageWriter
+}
+
+// WriteRobotsTxt writes the robots.txt content to the response writer.
+func (s *staticPageWriter) WriteRobotsTxt(rw http.ResponseWriter) {
+	s.writePage(rw, robotsTxtName)
+}
+
+// writePage writes the content of the page to the response writer.
+func (s *staticPageWriter) writePage(rw http.ResponseWriter, pageName string) {
+	content, ok := s.pages[pageName]
+	if !ok {
+		// If the page isn't regiested, something went wrong and there is a bug.
+		// Tests should make sure this code path is never hit.
+		panic(fmt.Sprintf("Static page %q not found", pageName))
+	}
+	_, err := rw.Write(content)
+	if err != nil {
+		logger.Printf("Error writing %q: %v", pageName, err)
+		s.errorPageWriter.WriteErrorPage(rw, http.StatusInternalServerError, "", err.Error())
```

Comment:
This will need a `req *http.Request` after my RequestID PR to extract out the {{.RequestID}} from the request.